systems/controllers/time_based_fsm.cc

```diff
@@ -0,0 +1,68 @@
+#include "systems/controllers/time_based_fsm.h"
+
+using std::cout;
+using std::endl;
+
+using std::string;
+using Eigen::VectorXd;
+using drake::systems::Context;
+using drake::systems::BasicVector;
+
+namespace dairlib {
+namespace systems {
+
+TimeBasedFiniteStateMachine::TimeBasedFiniteStateMachine(
+    RigidBodyTree<double>* tree,
+    int first_state_number,
+    int second_state_number,
+    int initial_state_number,
+    double duration_per_state,
+    double time_shift) :
+  first_state_number_(first_state_number),
+  second_state_number_(second_state_number),
+  initial_state_number_(initial_state_number),
+  duration_per_state_(duration_per_state),
+  time_shift_(time_shift) {
+  // Input/Output Setup
+  state_port_ = this->DeclareVectorInputPort(
+                  OutputVector<double>(tree->get_num_positions(),
+                                       tree->get_num_velocities(),
+                                       tree->get_num_actuators())).get_index();
+  this->DeclareVectorOutputPort(BasicVector<double>(1),
+                                &TimeBasedFiniteStateMachine::CalcFiniteState);
+}
+
+void TimeBasedFiniteStateMachine::CalcFiniteState(
+    const Context<double>& context,
+    BasicVector<double>* fsm_state) const {
+  // Read in current state and simulation time
+  const OutputVector<double>* robot_output = (OutputVector<double>*)
```

Comment:
I use robot_output to get the current time, but I can also get the time from context, so I will delete the port.

Reply:
I noticed that the robot walks better when it's using lcm time, because context time is slower than lcm time. I think I'll retain the state port for now.

Reply:
Leave for now, likely related to https://github.com/DAIRLab/dairlib/issues/63